src/types/my-resources/interfaces/myResources.interface.ts

```diff
@@ -18,3 +25,44 @@ export interface UpdateResourceCategory {
 export interface GetResourcesCategoriesParams extends Partial<RequestParams> {
   name?: string
 }
+
+export interface Lesson extends CommonEntityFields {
+  title: string
+  category: string | null | Category
+  text?: string
+  answers?: Answer[]
+  author?: string
+}
+
+export interface Quiz extends Question, CommonEntityFields {
+  title: string
+  // category: string | null
+  description: string
+  items: Question[]
+}
+
+export interface LessonData {
+  _id?: string
+  title: string
+  description: string
+  content: string
+  attachments: Attachment[]
+  category: string | null
+}
+
+export interface CourseResources extends CommonEntityFields {
+  title: string
+  text: string
+  answers: Answer[]
```

Comment:
можна було б додати тип null щоб передбачити відсутність відповідей:
```suggestion
  answers: Answer[] | null
```